File: webroot/cgi-bin/calculator.py
# {%ONLY POST%}
import sys
from PyCgiTool.ArgTool import ArgCgi


class CalculatorCgi(ArgCgi):
    def handle(self) -> None:
        try:
            first = float(self['first'])
            second = float(self['second'])
            rule = self['rule']
            res = 'Error!'
            if rule == 'add':
                res = first + second
            elif rule == 'multi':
                res = first * second
            self.add_output('result', str(res))
        except Exception as e:
            self.add_output('result', str(e))
        self.add_output('tool', 'result by python')


if __name__ == '__main__':
    CalculatorCgi(sys.argv).run()

File: webroot/cgi-bin/calculator.pl
# {%ONLY POST%}
my $arg = $ARGV[0];
my $li;
# 使用 split 函数将字符串分割成键值对

my @pairs = split /&/, $arg;
foreach my $pair (@pairs) {
    my ($key, $value) = split /=/, $pair;
    $li{$key} = $value;
}
my $num1 = int($li{'first'});
my $num2 = int($li{'second'});
my $op = int($li{'rule'});
my $res;
if($op eq 'add'){
    $res = $num1 + $num2;
}
else{
    $res = $num1 * $num2;
}
# 输出
print "{\"result\":\"$res\",\"tool\":\"result by perl\"}\n";
